ui/src/containers/VisualiseForm/StatementsForm/StatementsForm.js

```diff
@@ -77,6 +84,35 @@ class StatementsForm extends Component {
     </select>
   )
 
+  renderTimezoneSelector = () => {
+    const axes = unflattenAxes(this.props.model);
+    const groupAxes = axes.getIn(['group', 'optionKey'], 'date');
+    const groupingHasTimezone = periodKeys.includes(groupAxes);
+
+    const axesxQueryHas$dte = has$dte(this.props.model.get('axesxQuery'));
+    const axesyQueryHas$dte = has$dte(this.props.model.get('axesyQuery'));
+    const filterHas$dte = has$dte(this.props.model.get('filters'));
+
+    const periodHasOffset = this.props.model.get('previewPeriod') === 'TODAY';
+
+    const shouldDisplay = groupingHasTimezone || axesxQueryHas$dte || axesyQueryHas$dte || filterHas$dte || periodHasOffset;
+
+    return shouldDisplay && (
+      <TimezoneSelector
+        value={this.props.model.get('timezone', null)}
+        onChange={value => this.props.updateModel({
+          schema: 'visualisation',
+          id: this.props.model.get('_id'),
+          path: 'timezone',
+          value,
+        })}
+        defaultOption={{
+          label: buildDefaultOptionLabel(this.props.orgTimezone),
```

Comment:
Again if this creating a component inside the render we need to avoid that.

Reply:
`buildDefaultOptionLabel` is a function converting a string to string
type is `(orgTimezone: string)=> string`

Reply:
Okay that's fine, wasn't sure with this one.